docs/manual/guide/services/HeaderFilters.md

```diff
@@ -0,0 +1,30 @@
+# Header Filters
+
+In Lagom you may add `HeaderFilter`s to your `ServiceDescriptor`. In a `HeaderFilter` you will usually handle protocol negotiation or authentication. 
+
+A single `HeaderFilter` implementation may transform a request leaving a client or entering a server and a response leaving a server and entering a client.
+
+@[user-agent-auth-filter](code/docs/services/UserAgentHeaderFilter.java)
+
+This [`UserAgentHeaderFilter`](api/index.html?com/lightbend/lagom/javadsl/api/security/UserAgentHeaderFilter.html) is the default `HeaderFilter` any Lagom service will use if none is specified. It uses a `ServicePrincipal` which identifies the client with the service name. This way a server may identify a caller client by the `User-Agent`. 
+
+In `UserAgentHeaderFilter` the code at `transformClientRequest` will be invoked when preparing a client invocation to add a `User-Agent` header if a [ServicePrincipal](#ServicePrincipal) was specified on the request. On the server end `transformServerRequest` will be used to read the `User-Agent` header and set that value as the request's [Principal](https://docs.oracle.com/javase/8/docs/api/java/security/Principal.html).
+
+Keep in mind that a HeaderFilter should not be used to alter the request/response processing, it is just a tool to adapt two ends. 
+
+## Header Filter Composition
+
+Each service `Descriptor` can only have on `HeaderFilter`. In order to use several filters at once you may compose them using `HeaderFilter.composite` which will return a `HeaderFilter` that chains all the `HeaderFilter` you composed. When composing the order is important so when sending data the filters of the composite are used in the order they were provided, and when receiving data the filters will be used in reverse order. So if we registered two verbose Filters Foo and Bar like this:
```

Comment:
A few typos:

- only have one**e** HeaderFilter
- that chains all the `HeaderFilter`**'s** you compose-d-
- When composing**,** the